caffe2/core/operator_c10wrapper.cc

```diff
@@ -0,0 +1,11 @@
+#include "caffe2/core/operator_c10wrapper.h"
+
+namespace caffe2 {
+
+CAFFE_DEFINE_REGISTRY(
```

Comment:
So two important questions:
1) why do we have a separate registry? In the current setting you do lookup in precisely the same way as the main registry of C2
2) how does device handling happen? right now you unconditionally dispatch to this registry without looking at DeviceType. Is the intention that with Tensor class merge we'd just dispatch on Tensor class itself and let your new dispatcher handle the details? If so - let's add those comment so that you diff doesn't get affected by the Tensor refactoring

Reply:
The new dispatcher runs at invocation time, not at op instantiation time. `TryCreateOperator()` just returns you a wrapper that will call the dispatcher whenever you invoke the op.

This is also the reason why this is a separate registry. In caffe2, there's different registries for different devices, but I want to ignore the device and always return this wrapper.

Reply:
Following up on this thread, I agree with Dima that we should not have separate op registries.

Reply:
How would that work? Caffe2 already has multiple registries, different ones for CPU and for CUDA for example. Since c10 ops handle both, CPU and CUDA, I can't use either of them. That's why I added a third one.